pkg/restclient/client.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 The Cloud Native Events Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	ce "github.com/cloudevents/sdk-go/v2/event"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+	log "github.com/sirupsen/logrus"
+
+	"golang.org/x/net/context"
+)
+
+var (
+	httpTimeout = 2 * time.Second
+)
+
+// Rest client to make http request
+type Rest struct {
+	client http.Client
+}
+
+// New get new rest client
+func New() *Rest {
+	return &Rest{
+		client: http.Client{
+			Timeout: httpTimeout,
+		},
+	}
+}
+
+// PostEvent post an event to the give url and check for error
+func (r *Rest) PostCloudEvent(url *types.URI, e ce.Event) (status int, err error) {
+	b, err := json.Marshal(e)
+	if err != nil {
+		log.Errorf("error marshalling event %v", e)
+		return status, err
+	}
+
+	if status = r.Post(url, b); status == http.StatusBadRequest {
```

Comment:
posting of cloud events should be wrapped with O-Ran cloudevents API,. Why rest API is posting cloudEvents object ?

Reply:
O-RAN spec uses Cloud Events as Event Data Model.

> 7.2 Status Notifications Data Model
> This clause specifies the event Status Notification data model supported by the API. The current model supports JSON encoding of the CloudEvents.io specification [15] for the event payload.